### ISSUE TITLE
Saving Custom Command response data to variables

### ISSUE THREAD

**Issue by tomdevlive** [bitfocus/companion-module-obs-studio#284]
I'm not sure if this already exists and I just haven't figured it out. When sending a Custom Command like GetSceneItemTransform, I would like to get positionX and positionY values from the response and save those to custom variables. I don't know where or how to get the response data. Is it there I just don't see it? If so could I do the 'set variable with expression' and pull it that way? Any help would be greatly appreciated.

**Comment by bryce-seifert** [comment]
This should now be possible in the latest Companion beta.

It adds a “custom_command_response” variable which shows the last reply from a custom command call, which you can use with jsonparse to set a variable with an expression, or a variety of other workflows. Hopefully this is helpful!

**Comment by tomdevlive** [comment]
@bryce-seifert Ok this looks pretty cool, however is it just the one response variable for all my custom commands? I have custom commands that change audio and custom commands that change transforms for scene items. Let's say I was wanting to get the response from a transform command but right at that moment an audio command happened? **Also please know I am in no way being ungrateful,** this is really awesome. I'm just trying to figure out how to incorporate it. If I wanted to for sure get specific response data from a specific custom command then I need to make sure no other custom commands run at all, except for the one I'm trying to get the response data from. Is there a way to get response data from a specific custom command, essentially ignoring all other custom commands? Like a way to tie different response variables to different custom commands,. so that I can run multiple custom command simultaneously and still collect the seperate response data from each?

**Comment by bryce-seifert** [comment]
@tomdevlive Good questions! Yes, it currently is only one variable response value. There's not a super intuitive way to allow users to generate variables within a module on demand in Companion. But, I included the `$(obs:custom_command_type)` in hopes that would help with parsing. I'm not sure how often you're triggering commands, or the volume, but it should be possible to setup a trigger that can save to a custom Companion variable based on the command type changing. I've attached my example here.

Obviously, this would be a bit cumbersome if you have more than a few of these, and I haven't tested it if you're sending tons of custom commands a second, so it may fall apart there.

![Image](https://github.com/user-attachments/assets/48411182-9e18-4e33-9ebe-4ac791ed21b3)
